quipucords/fingerprinter/runner.py

```diff
@@ -1015,21 +1005,25 @@ def _process_network_fact(self, source: dict, fact: dict) -> dict:
         fingerprint = {META_DATA_KEY: {}}
 
         # Use a list to drive direct mappings of raw_fact_key, fingerprint_key pairs.
-        raw_fact_keys_to_fingerprint_keys = [
+        raw_fact_fingerprint_fn = [
             # Common facts
-            ("uname_hostname", "name"),
-            ("uname_processor", "architecture"),
+            ("uname_hostname", "name", formatters.str_or_none),
+            ("uname_processor", "architecture", formatters.str_or_none),
             # Red Hat facts
-            ("redhat_packages_gpg_num_rh_packages", "redhat_package_count"),
-            ("redhat_packages_certs", "redhat_certs"),
-            ("redhat_packages_gpg_is_redhat", "is_redhat"),
-            ("etc_machine_id", "etc_machine_id"),
+            (
+                "redhat_packages_gpg_num_rh_packages",
+                "redhat_package_count",
+                formatters.int_or_none,
+            ),
+            ("redhat_packages_certs", "redhat_certs", formatters.str_or_none),
+            ("redhat_packages_gpg_is_redhat", "is_redhat", convert_to_bool_or_none),
+            ("etc_machine_id", "etc_machine_id", formatters.str_or_none),
             # Set OS information
-            ("etc_release_name", "os_name"),
-            ("etc_release_version", "os_version"),
-            ("etc_release_release", "os_release"),
+            ("etc_release_name", "os_name", formatters.str_or_none),
+            ("etc_release_version", "os_version", formatters.str_or_none),
+            ("etc_release_release", "os_release", formatters.str_or_none),
             # Installed products (name + product/eng ID pairs)
-            ("installed_products", "installed_products"),
+            ("installed_products", "installed_products", formatters.str_or_none),
```

Comment:
Since this is `JSONField` in DB, should we have a formatter that also verifies that string can be parsed as JSON?

Reply:
TY. Camayoc is correct! installed products is indeed a list of dicts. Will fix this and add a unit test covering this one